crates/networking/p2p/peer_handler.rs

```diff
@@ -58,6 +58,13 @@ impl PeerHandler {
     pub fn new(peer_table: Arc<Mutex<KademliaTable>>) -> PeerHandler {
         Self { peer_table }
     }
+
+    /// Creates a dummy PeerHandler for tests where interacting with peers is not needed
+    /// This should only be used in tests as it won't be able to interact with the node's connected peers
+    pub fn dummy() -> PeerHandler {
```

Comment:
nit: keep this inside test flag?

Reply:
I would like to but the `test` feature doesn't behave well when importing across crates